pkg/devfile/devfile.go

```diff
@@ -0,0 +1,25 @@
+package devfile
```

Comment:
Are we sure it makes sense to include packages from outside `api/...` here? This removes the devfile header from versioning. This makes sense since `devfile` isn't a CRD, so maybe the entire `devfile.go` file should be moved to `pkg/devfile` and import `DevWorkspaceTemplateSpec`.

Reply:
> This makes sense since devfile isn't a CRD, so maybe the entire `devfile.go` file should be moved to `pkg/devfile` and import `DevWorkspaceTemplateSpec`

Well, since the `DevWorkspaceTemplateSpec` should in any case be part of the K8S API, importing it from outside the K8S API package, doing this would make the `common` package depend on a given K8S API version, which is precisely what we want to avoid.

The new `DevfileHeader` structure (which contains only `schemaVersion` and `DevfileMetadata`) is the only thing in the whole model that is not in any way coupled with (or used in) the versioned K8S API. On the contrary the main Devfile body structure (`DevWorkspaceTemplateSpec`) is also part of the K8S API.